File: android_mobile/src/main/java/com/alexstyl/specialdates/contact/NameComparator.java
package com.alexstyl.specialdates.contact;

import com.alexstyl.gsc.SoundComparer;

class NameComparator {

    public static final NameComparator INSTANCE = new NameComparator();

    private NameComparator() {
        // hide this
    }

    public boolean areTheSameName(String o1, String o2) {
        return SoundComparer.areSame(o1, o2);
    }
}

File: android_mobile/src/main/java/com/alexstyl/specialdates/contact/NameComparator.kt
package com.alexstyl.specialdates.contact

import com.alexstyl.gsc.SoundComparer

object NameComparator {

    fun areTheSameName(left: String, right: String): Boolean {
        try {
            return SoundComparer.areSame(left, right)
        } catch (e: Exception) {
            return false
        }

    }

}
